xds/src/main/java/io/grpc/xds/internal/sds/CommonTlsContextUtil.java

```diff
@@ -37,7 +37,18 @@ private static boolean hasCertProviderValidationContext(CommonTlsContext commonT
           commonTlsContext.getCombinedValidationContext();
       return combinedCertificateValidationContext.hasValidationContextCertificateProviderInstance();
     }
-    return commonTlsContext.hasValidationContextCertificateProviderInstance();
+    return hasValidationProviderInstance(commonTlsContext);
+  }
+
+  private static boolean hasIdentityCertificateProviderInstance(CommonTlsContext commonTlsContext) {
+    return commonTlsContext.hasTlsCertificateProviderInstance()
+        || commonTlsContext.hasTlsCertificateCertificateProviderInstance();
+  }
+
+  private static boolean hasValidationProviderInstance(CommonTlsContext commonTlsContext) {
+    return (commonTlsContext.hasValidationContext() && commonTlsContext.getValidationContext()
+        .hasCaCertificateProviderInstance())
+        || commonTlsContext.hasValidationContextCertificateProviderInstance();
```

Comment:
If this method `hasValidationProviderInstance` is only used once, why not use it inline?

Reply:
Similar to other functions defined here e.g. `hasIdentityCertificateProviderInstance` or `hasCertProviderValidationContext`. If I inline all functions used only once I'll end up with complex functions (with successive such inlining). In general if it helps in understanding the code I think it is good to have separate, appropriately named functions. The compiler will anyway inline such functions so I don't see the downside.